apps/docs/src/content/examples/shaders/TSL Slice Material.mdx

```diff
@@ -0,0 +1,76 @@
+The example below is an adaptation of [this](https://threejs.org/examples/?q=tsl#webgpu_tsl_angular_slicing) threejs example.
+
+<Example path="shaders/slice/main" />
+
+A question that comes up often in ThreeJS development is "How do I extend ThreeJS's materials?". Most libraries such as [three-custom-shader-material](https://www.npmjs.com/package/three-custom-shader-material) use a find and replace solution. This works well enough but ThreeJS has identified that it's not an ideal solution and recommends using the [threejs shading language](https://github.com/mrdoob/three.js/wiki/Three.js-Shading-Language). One of the primary benefits for using TSL is that the language is renderer-agnostic. This means TSL will compile to either glsl or wgsl code depending on what renderer is being used.
+
+<Tip type="warning">
+  At this time webgpu is still a very experimental browser api and is [not widely
+  supported](https://caniuse.com/?search=webgpu). For this reason, there are ways to fallback to
+  webgl when webgpu is not available.
+</Tip>
+
+Threlte easily integrates with node materials from `three/webgu`. The first step is to switch to using a `WebGPURenderer`.
+
+```svelte
+<Canvas
+  createRenderer={(canvas) => {
+    return new WebGPURenderer({
+      antialias: true,
+      canvas,
+      forceWebGL: false
+    })
+  }}
+>
+  <Scene />
+</Canvas>
+```
+
+The next step is to extend the threlte catalog to use the definitions from `three/webgu`.
+
+```svelte
+<script>
+  import * as THREE from 'three/webgpu'
+  import { extend } from '@threlte/core'
+
+  extend(THREE)
+</script>
+```
+
+The extension makes it possible to use class definitions such as `MeshPhysicalNodeMaterial` on `<T>` components.
+
+```svelte
+<T.MeshPhysicalNodeMaterial />
+```
```

Comment:
This section is very interesting and likely worthwhile when generally using `three/webgpu`. Do you think it makes sense to perhaps move it to the webgpu renderer example and link a "tips for getting started with webgpu"? Or alternatively copy this section for that example as well?

Reply:
Another random thought that I had when reading this over that is definitely outside the scope of this PR but is a bit of a looming problem: 

Three.js's recent change to bundle `three/webgpu` separately will probably pose big tree shaking issues for us as its adoption increases since the T component does a wildcard import of the `three` package. So here we're actually wildcard importing both `three` and `three/webgpu`, which I bet will result in large bundle sizes.

Reply:
> This section is very interesting and likely worthwhile when generally using `three/webgpu`. Do you think it makes sense to perhaps move it to the webgpu renderer example and link a "tips for getting started with webgpu"? Or alternatively copy this section for that example as well?

i agree. i wonder if we should have a webgpu section that has a /examples/webgpu/getting-started index page or something along those lines.

Reply:
> Another random thought that I had when reading this over that is definitely outside the scope of this PR but is a bit of a looming problem:
> 
> Three.js's recent change to bundle `three/webgpu` separately will probably pose big tree shaking issues for us as its adoption increases since the T (`T.ts` in core) component does a wildcard import of the `three` package. So here we're actually wildcard importing both `three` and `three/webgpu`, which I bet will result in large bundle sizes.

yeah you can see the console logs out that two instances of three are being used. same with the other webgpu example. i'm not sure how to consolidate that.

Reply:
I'm definitely unsure as well. First thought is that we may have to introduce a `@threlte/core/webgpu` export with a different internal three import. Not sure yet, but we've got time to think!

Reply:
Pretty sure three.js team is still trying to figure out their best route too: https://github.com/mrdoob/three.js/pull/29404